masonry/src/widget/progress_bar.rs

```diff
@@ -174,6 +230,41 @@ impl Widget for ProgressBar {
         smallvec![self.label.id()]
     }
 
+    fn on_timer_expired(&mut self, ctx: &mut EventCtx, _event: &TimerEvent) {
+        tracing::debug!("progress_bar: on_timer_expired");
+        self.reset_animation();
+        ctx.request_anim_frame();
+        ctx.request_paint_only();
+    }
+
+    fn on_anim_frame(&mut self, ctx: &mut UpdateCtx, interval: u64) {
+        tracing::debug!("progress_bar: on_anim_frame");
+        // pixel per 1ms
+        const NS_PER_PIXEL: f64 = 1_000_000.;
+        const DURATION_BETWEEN_ANIMATIONS: Duration = Duration::from_millis(2_500);
```

Comment:
It's not super clear to me why you want the progress bar to have an animation every 2.5 seconds?

Reply:
It's arbitrary - I'm copying the way progress bars work on windows (with periodic movement to indicate progress). My motivation for having timers is triggering events at specific intervals in a simple way without needing to feed in messages from external tasks etc. Happy to go with whatever setup we wanted for animation on the progress bar. In any case it should be optional, with a `with_animation` on at least the xilem view. If I didn't do this already I should before merging.